### PR TITLE
WINDUP-4149: MTR 1.2.5 Release Notes

### PR DESCRIPTION
**Pull request by anarnold97**
### JIRA

* [WINDUP-4149](https://issues.redhat.com/browse/WINDUP-4149)

### PREVIEW 

* [MTR 1.2.5](https://deploy-preview-868--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-1-2-5)

    * [New features](https://deploy-preview-868--windup-documentation.netlify.app/docs/release-notes-mtr/master/#rn-new-features-1-2-5_release-notes)
    * [Known issues](https://deploy-preview-868--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-rn-known-issues-1-2-5_release-notes)
    * [Resolved issues](https://deploy-preview-868--windup-documentation.netlify.app/docs/release-notes-mtr/master/#mtr-rn-resolved-issues-1-2-5_release-notes)

### CODE REVIEW

**Comment by PhilipCattanach** [comment]
A few comments. I would not say it is a container grade only release as there are some rules changes and some updates to dependencies to fix CVEs. Rather it is a bug fix release with 2 new rulesets.
So I would call out under new features New EAP XP5 rules to highlight that Microprofile Metrics and Microprofile OpenTracing are not supported..
Also the Resolved Issue filter I think should just seslect issues of Type Bug and with a fixVersion of MTR-1.2.5
Otherwise it includes all of the tasks associated with doing the release, which is not what a customer would be interested in.

File: docs/topics/mtr-rn-new-features-1-2-5.adoc [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+//
+// * docs/release_notes/master.adoc
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-5_{context}"]
+= New features
+
+// This section describes the new features of the {ProductName} ({ProductShortName}) 1.2.5:
+
+{ProductName} ({ProductShortName}) 1.2.5 is a container grade only release. There are no new features.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-5.adoc:11]:
```suggestion
{ProductName} ({ProductShortName}) 1.2.5 has no new features.
```

File: docs/topics/mtr-rn-new-features-1-2-5.adoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+//
+// * docs/release_notes/master.adoc
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-5_{context}"]
+= New features
+
+// This section describes the new features of the {ProductName} ({ProductShortName}) 1.2.5:
+
+{ProductName} ({ProductShortName}) 1.2.5 has no new features.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-5.adoc:11]:
```suggestion
{ProductName} ({ProductShortName}) 1.2.5 has the following new features:
```

File: docs/topics/mtr-rn-new-features-1-2-5.adoc [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+// * docs/release_notes/master.adoc
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-5_{context}"]
+= New features
+
+// This section describes the new features of the {ProductName} ({ProductShortName}) 1.2.5:
+
+{ProductName} ({ProductShortName}) 1.2.5 has the following new features:
+
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-5.adoc:12]:
```suggestion



.New ruleset for MP metrics replaces old ruleset

A new ruleset for MicroProfile (MP) metrics replaces the old ruleset. link:https://issues.redhat.com/browse/WINDUPRULE-1043[(WINDUPRULE-1043)]

.New ruleset for  MP OpenTracing replaces the old ruleset

A new ruleset for MicroProfile (MP) OpenTracing replaces the old ruleset. link:https://issues.redhat.com/browse/WINDUPRULE-1044[(WINDUPRULE-1044)]
```

File: docs/topics/mtr-rn-resolved-issues-1-2-5.adoc [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-1300[(CVE-2024-1300)].
+
+
+.New ruleset for MP metrics replaces old ruleset
+
+A new ruleset for MicroProfile (MP) metrics replaces the old ruleset. link:https://issues.redhat.com/browse/WINDUPRULE-1043[(WINDUPRULE-1043)]
+
+.New ruleset for  MP OpenTracing replaces the old ruleset
+
+A new ruleset for MicroProfile (MP) OpenTracing replaces the old ruleset. link:https://issues.redhat.com/browse/WINDUPRULE-1044[(WINDUPRULE-1044)]
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-5.adoc:36]:
```suggestion

```

File: docs/topics/mtr-rn-new-features-1-2-5.adoc [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+.New ruleset for MP metrics replaces old ruleset
+
+A new ruleset for MicroProfile (MP) metrics replaces the old ruleset. link:https://issues.redhat.com/browse/WINDUPRULE-1043[(WINDUPRULE-1043)]
+
+.New ruleset for  MP OpenTracing replaces the old ruleset
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-5.adoc:17]:
```suggestion
.New ruleset for MicroProfile OpenTracing replaces the old ruleset
```

File: docs/topics/mtr-rn-new-features-1-2-5.adoc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+:_content-type: CONCEPT
+[id="rn-new-features-1-2-5_{context}"]
+= New features
+
+// This section describes the new features of the {ProductName} ({ProductShortName}) 1.2.5:
+
+{ProductName} ({ProductShortName}) 1.2.5 has the following new features:
+
+.New ruleset for MP metrics replaces old ruleset
```
Comment (anarnold97) [inline docs/topics/mtr-rn-new-features-1-2-5.adoc:13]:
```suggestion
.New ruleset for MicroProfile metrics replaces old ruleset
```

**Comment by PhilipCattanach** [comment]
Thanks @anarnold97 the release notes look fine. The Resolved Issues filter needs to be updated but obviously that isn't a docs change, rather a Jira filter change.

File: docs/topics/mtr-rn-resolved-issues-1-2-5.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-26308[(CVE-2024-26308)].
+
+.CVE-2024-1300: A vulnerability in the Eclipse Vert.x toolkit causes a memory leak in TCP servers configured with TLS and SNI support
+
+A vulnerability in the Eclipse Vert.x toolkit causes a memory leak in Transmission Control Protocol (TCP) servers configured with TLS and SNI support. When processing an unknown Server Name Indication (SNI) server name assigned the default certificate instead of a mapped certificate, the Secure Sockets Layer (SSL) context is erroneously cached in the server name map, leading to memory exhaustion. This affects only TLS servers with SNI enabled. Users are recommended to upgrade to {ProductShortName} 1.2.5, which resolves this issue.
+
+For more details, see link:https://access.redhat.com/security/cve/CVE-2024-1300[(CVE-2024-1300)].
+
+
+For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12432806[MTR 1.2.5 resolved issues] in Jira.
```
Comment (anarnold97) [inline docs/topics/mtr-rn-resolved-issues-1-2-5.adoc:30]:
```suggestion
// filter changed to:
// project in (WINDUP, WINDUPRULE) AND issuetype = Bug AND fixVersion = MTR-1.2.5 ORDER BY  id

For a complete list of all issues resolved in this release, see the list of link:https://issues.redhat.com/issues/?filter=12434315[MTR 1.2.5 resolved issues] in Jira.
```

**Comment by anarnold97** [comment]
@PhilipCattanach - updated the filter. it is a weird one as it shows CVEs which only RH employees can view
Thanks

**Comment by PhilipCattanach** [comment]
I see @anarnold97 - So it's not a permissions problem with the Jira filter itself. Rather all of the issues that is is selecting are Security issues. That are locked down.

**Comment by anarnold97** [comment]
@PhilipCattanach 
Aye it looks grand if you work for RH employees
And honestly, I still see value in that, so I vote to keep it

**Pull request merged**